Fix stlinkv3 platform bits

ALTracer opened the pull request:
## Detailed description

* There is a small new feature.
* Existing problems are `(gdb) compare-sections` potentially not working on STLINK/V3, and SWO capture baudrate too low.
* This PR solves these by enabling RCC_CRC and switching UART5 clocksource from APB1 Pclk to Sysclk.

Also I add a couple paragraphs of documentation to platform README.md with info from ST manuals and libopencm3.
Testing on stlinkv3 required prior to merge.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (see [Building the firmware](https://github.com/blackmagic-debug/blackmagic?tab=readme-ov-file#building-black-magic-debug-firmware)) *and does not affect it*
* [x] It builds as BMDA (see [Building the BMDA](https://github.com/blackmagic-debug/blackmagic?tab=readme-ov-file#building-black-magic-debug-app)) *and does not affect it*
* [ ] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

## Closing issues
None reported.

dragonmux reviewed (approved):
This all looks reasonable, so as long as someone can vouch it works, we're happy to merge it. Thanks for the contribution!

ALTracer commented:
The CRC patch is good, and now I've tested it -- a positive change from bogus `C0` replies to actually matching `compare-sections` at highest speeds (150..220..390 KiB/s with `s/-Og/-Os/`). Thanks to Uwe Bonnes for noticing it.
The clocksource select change is promising, but on RDP2 chips inception debugging is impossible, and the DEBUG_INFO I added in `platform_init()` has no chance to run and be useful. I dropped the logline and corresponding readouts.
Thinking further, why bump only SWO UART? I bumped Aux UART in a similar way. I'd need `usart_get_baudrate()` small function written/ported to verify operation -- it does report some discrete baudrates, which also are pretty high, but libopencm3 API doesn't limit BRR to 16 or more, so some of the numbers are bogus (like a 216M baudrate with a prescaler of 1+0). I'll report later when I'll be able to test high baudrates against an f411ce or FT4232H or some other fast MCU (or even a MSO) -- but of course independent testing is welcome.

dragonmux reviewed (approved):
Given you've told us this fixes a bug/issue, we'll get this merged anyway. If some part of the speed enhancements here do actually cause problems, they can be adjusted again later.
